api/auth_middleware.go

```diff
@@ -24,57 +34,135 @@ func getServiceAccountID(ctx context.Context) (string, bool) {
 }
 
 // authMiddleware authenticates either access_token or key pair
-func authMiddleware(
-	sasUC usecases.ServiceAccounts,
-) func(http.Handler) http.Handler {
+func authMiddleware(sasUC usecases.ServiceAccounts) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			authorization := r.Header.Get("authorization")
-			parts := strings.Split(authorization, " ")
-			if authorization == "" || len(parts) != 2 {
-				w.WriteHeader(http.StatusUnauthorized)
+		return http.HandlerFunc(func(responseWriter http.ResponseWriter, request *http.Request) {
+			header := request.Header.Get("authorization")
+			authHeaderContents := strings.Split(header, " ")
+			authHeader := buildAuth(header, authHeaderContents[0], authHeaderContents[1])
+
+			if !authHeader.isValid() {
+				responseWriter.WriteHeader(http.StatusUnauthorized)
 				return
 			}
+
```

Comment:
Agree with you about the large functions (which I think its not the case here) 👍
In my experience, using empty lines to divide functions into chunks of related logic can help to increase its readability, as the flow is now more or less divided, instead of being a single blob of code.